python/pyspark/ml/deepspeed/deepspeed_distributor.py

```diff
@@ -97,6 +98,8 @@ def __init__(
             _ssl_conf=DeepspeedTorchDistributor._DEEPSPEED_SSL_CONF,
         )
         self.cleanup_deepspeed_conf = False
+        if not useGpu:
+            warnings.warn(message="Deepspeed usage with CPUs is not very well supported at this time.")
```

Comment:
Just say "DeepspeedDistributor usage with CPUs is not supported at this time".

Reply:
We don't want approximate phrases like "not very well"

Reply:
We may want to raise an error actually

Reply:
we want the phrase to be more of a warning rather than a flat statement. Technically, I think Deepspeed supports *some* CPU use cases but it's very unstable + flaky. If the user knows their program will work, we might as well let them try it right?

Reply:
But we haven't successfully tested it in our use case, right? If they do run into errors, they will probably message oncall and we don't have a plan to approach such fixes. I will let @lu-wang-dl make the final decision. Should we raise a warning or an error?

Reply:
I agree we should raise an error saying that CPUs are not supported currently